subsys/bluetooth/host/iso.c

```diff
@@ -2258,7 +2258,7 @@ static int hci_le_create_big(struct bt_le_ext_adv *padv, struct bt_iso_big *big,
 	req->framing = param->framing;
 	req->encryption = param->encryption;
 	if (req->encryption) {
-		memcpy(req->bcode, param->bcode, sizeof(req->bcode));
+		sys_memcpy_swap(req->bcode, param->bcode, sizeof(req->bcode));
```

Comment:
```suggestion
		memcpy(req->bcode, param->bcode, sizeof(req->bcode));
```

Reply:
So you don't think we should allow applications to set e.g. "Broadcast Code" as the broadcast code, but rather have them do "\0\0edoC tsacdaorB"?

Reply:
Rather application do assign "Broadcast Code\0\0", where `B` will be least significant byte in the `bcode` uint8_t array on a little endian machine, and most significant bytes are padding "\0\0" (as required by the BT Spec.)

Reply:
@cvinayak Is correct. And it's correct independent of the machine endianness. That's because the `bcode` array is always a little-endian serialization of a uint_128 (so not always machine endianness), and UTF-8 does not have endianness. So it's always "first byte of string goes in first byte of `bcode`".

Reply:
And, if the application wants to supply a number instead of a string, then the application must convert the number to a LE byte array with padding.

Reply:
Re-reading it today with fresh eyes makes it clearer to me that you are right :) 

In the statement
> “Børne House” is represented as the value 0x00000000_6573756F_4820656E_72B8C342.

"B" (or 42) is indeed the east significant byte of that value, and should be place at bcode[0].

Reply:
> By the way, the '\0\0' is not needed when the string literal is used as an initializer. The C standard guarantees that elements missing from initializers are zeroed.
> 
> [ISO/IEC 9899:1999](http://www.open-std.org/jtc1/sc22/WG14/www/docs/n1256.pdf), 6.7.8, 21:
> 
> > If there are fewer initializers in a brace-enclosed list than there are elements or members of an aggregate, or fewer characters in a string literal used to initialize an array of known size than there are elements in the array, the remainder of the aggregate shall be initialized implicitly the same as objects that have static storage duration.

That is true, if it is used as an initializer. 

If the code was something along the lines of
```
const char *code_string = "Broadcast Code";
memcpy(param.bcode, code_string, sizeof(code_string));
```
then it wouldn't work as intended.